Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-                m_LastLogMove = -1;
-            }
-        }
-    }
-
-// TODO: This sometime.. bleh
-    // Cut off the text in the text label at a reasonable height so it doesn't get really slow to draw
-//    if (m_pConsoleText->GetTextHeight() > g_FrameMan.)
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          Draw
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Draws this ConsoleMan's current graphical representation to a
-//                  BITMAP of choice.
-
-void ConsoleMan::Draw(BITMAP *pTargetBitmap)
-{
-    if (IsVisible() || m_ForceVisible)
-    {
-        AllegroScreen drawScreen(pTargetBitmap);
-        m_pGUIController->Draw(&drawScreen);
-    }
-}
-
-
-//////////////////////////////////////////////////////////////////////////////////////////
-// Method:          ClearLog
-//////////////////////////////////////////////////////////////////////////////////////////
-// Description:     Clears all previous input.
-
-void ConsoleMan::ClearLog()
-{
-    m_InputLog.clear();
-    m_InputLogPosition = m_InputLog.begin();
-    m_pConsoleText->SetText("");
-}
-
-
-} // namespace RTE
+
+		m_InputTextBox->SetText("");
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::ConsoleOpenClose() {
+		float toGo = 0;
+		switch (m_ConsoleState) {
+			case ENABLING:
+				m_ParentBox->SetEnabled(true);
+				m_ParentBox->SetVisible(true);
+
+				toGo = -std::floorf(static_cast<float>(m_ParentBox->GetYPos()) * 0.5F);
+				m_ParentBox->SetPositionAbs(0, m_ParentBox->GetYPos() + toGo);
+
+				if (m_ParentBox->GetYPos() >= 0) { m_ConsoleState = ENABLED; }
+				break;
+			case ENABLED:
+				// If supposed to be enabled but appears to be disabled, enable it gracefully
+				if (m_ParentBox->GetYPos() < 0) { m_ConsoleState = ENABLING; }
+
+				m_ParentBox->SetEnabled(true);
+				m_ParentBox->SetVisible(true);
+				break;
+			case DISABLING:
+				toGo = -std::ceilf((static_cast<float>(m_ParentBox->GetHeight()) + static_cast<float>(m_ParentBox->GetYPos())) * 0.5F);
+				m_ParentBox->SetPositionAbs(0, m_ParentBox->GetYPos() + toGo);
+
+				if (m_ParentBox->GetYPos() <= -m_ParentBox->GetHeight()) {
+					m_ParentBox->SetEnabled(false);
+					m_ParentBox->SetVisible(false);
+					m_ConsoleState = DISABLED;
+				}
+				break;
+			case DISABLED:
+				// If supposed to be disabled but appears to be out anyway, disable it gracefully
+				if (m_ParentBox->GetYPos() > -m_ParentBox->GetHeight()) { m_ConsoleState = DISABLING; }
+
+				m_ParentBox->SetEnabled(false);
+				m_ParentBox->SetVisible(false);
+				m_InputTextBox->SetEnabled(false);
+				m_GUIController->GetManager()->SetFocus(0);
+				break;
+			default:
+				RTEAbort("Undefined console state value passed in. See ConsoleState enumeration for defined values.");
```

Comment:
You should specify what state (even if it's just a number) was passed in in the abort message. Not like this'll ever happen, but it can make things a little easier to track down.